web/src/components/vis/analyses.js

```diff
@@ -164,3 +164,14 @@ export default [
     icon: vuetify.icons.pca,
   },
 ];
+
+// Create a path-indexable map version of the analysis list.
+const analysisMap = Object.freeze(analysisList.reduce((map, entry) => ({
+  ...map,
+  [entry.path]: entry,
+}), {}));
```

Comment:
@waxlamp while subtle, this results in something rather different than what I was suggesting. 

* a new object is created at every iteration of the loop.  This is unnecessary, since reduce() takes an initializer as its second param and you can use the same object for the whole loop.  This is at worst an `O(N)` cost
* somewhat more tricky, though, is that spread.  You can never really know what babel will do with language features like this, but in this case, I think it gets transpiled into an `Object.keys()` call.  What would have been a `O(n)` loop is now `O(n*2 * m * c)` where N is array length, M is keys per object, and C is some constant because it looks like the code babel generates does a bunch of looping and filtering.

I wouldn't have bothered to mention this, but I'd regret if this code to get copied somewhere else and used under different circumstances.

[Here's the worst-case scenario](https://babeljs.io/repl#?browsers=defaults%2C%20not%20ie%2011%2C%20not%20ie_mob%2011&build=&builtIns=false&spec=false&loose=false&code_lz=MYewdgzgLgBAhjAvDA3gXwNwChSVgIyVQDpS5Mg&debug=false&forceAllTransforms=false&shippedProposals=false&circleciRepo=&evaluate=false&fileSize=false&timeTravel=false&sourceType=module&lineWrap=true&presets=env%2Creact%2Cstage-2%2Cenv&prettier=false&targets=&version=7.11.4&externalPlugins=)

Reply:
Thanks for pointing this out (and for not explicitly calling me out for trying to be too clever :stuck_out_tongue:). I'll open a PR to fix this.